builder.go

```diff
@@ -181,6 +206,15 @@ func (b *Builder) Build() (*BuildMetadata, error) {
 	}, nil
 }
 
+func withBuildpack(bp Buildpack, bom []BOMEntry) []BOMEntry {
```

Comment:
Maybe this should be a function on `[]BOMEntry` so that it reads `bom.withBuildpack(bp)`